src/D2L.CodeStyle.TestAnalyzers/NUnit/AssertIsBool/AssertIsBoolBinaryExpressions.cs

```diff
@@ -0,0 +1,270 @@
+
+				{ SyntaxKind.LessThanToken, m_lessThanDiagnosticProviderFactory },
+				{ SyntaxKind.GreaterThanToken, e => m_lessThanEqualsDiagnosticProviderFactory( e ).Opposite() },
+
+				{ SyntaxKind.LessThanEqualsToken, m_lessThanEqualsDiagnosticProviderFactory },
+				{ SyntaxKind.GreaterThanEqualsToken, e => m_lessThanDiagnosticProviderFactory( e ).Opposite() },
+
+				{ SyntaxKind.IsKeyword, m_isKeywordDiagnosticProviderFactory }
+			}.ToImmutableDictionary();
+
+		public static bool TryGetDiagnosticProvider(
+			InvocationExpressionSyntax invocation,
+			out AssertIsBoolDiagnosticProvider diagnosticProvider
+		) {
+			if( invocation.ArgumentList.Arguments.Count == 0 ) {
+				diagnosticProvider = null;
+				return false;
+			}
+
+			if( !( invocation.ArgumentList.Arguments[ 0 ].Expression is BinaryExpressionSyntax binaryExpression ) ) {
+				diagnosticProvider = null;
+				return false;
+			}
+
+			bool isKnownExpression = m_diagnosticProviders.TryGetValue(
+				binaryExpression.OperatorToken.Kind(),
+				out Func<InvocationExpressionSyntax, AssertIsBoolDiagnosticProvider> diagnosticProviderFactory
+			);
+
+			if( !isKnownExpression ) {
+				diagnosticProvider = null;
+				return false;
+			}
+
+			diagnosticProvider = diagnosticProviderFactory( invocation );
+			return true;
+		}
+
+		private static AssertIsBoolDiagnostic GetComparisonDiagnostic(
+			InvocationExpressionSyntax invocation,
+			string replacementMethodName
+		) {
+			BinaryExpressionSyntax binaryExpression = (BinaryExpressionSyntax) invocation.ArgumentList.Arguments[ 0 ].Expression;
+
+			return GetDiagnostic( 
+					invocation,
+					SyntaxFactory.IdentifierName( replacementMethodName ),
+					binaryExpression.Left,
+					binaryExpression.Right
+				);
+		}
+
+		private static AssertIsBoolDiagnostic GetEqualityOperatorsDiagnostic(
+			InvocationExpressionSyntax invocation,
+			string comparisonReplacementMethodName,
+			string nullReplacementMethodName,
+			string zeroReplacementMethodName,
+			string trueReplacementMethodName,
+			string falseReplacementMethodName
+		) {
+			BinaryExpressionSyntax binaryExpression = (BinaryExpressionSyntax) invocation.ArgumentList.Arguments[ 0 ].Expression;
+
+			string replacementMethodName;
+			ExpressionSyntax[] firstParameterReplacements;
+
+			bool IsNullLiteral( ExpressionSyntax expression ) =>
+				expression.Kind() == SyntaxKind.NullLiteralExpression;
+
+			bool IsZeroLiteral( ExpressionSyntax expression ) =>
+				expression.Kind() == SyntaxKind.NumericLiteralExpression && expression.ToString() == "0";
+
+			bool IsTrueLiteral( ExpressionSyntax expression ) =>
+				expression.Kind() == SyntaxKind.TrueLiteralExpression;
+
+			bool IsFalseLiteral( ExpressionSyntax expression ) =>
+				expression.Kind() == SyntaxKind.FalseLiteralExpression;
+
+			ExpressionSyntax otherOperand;
+			if( binaryExpression.TryGetSingleOperandForEqualityReplacement( IsNullLiteral, out otherOperand ) ) {

```

Comment:
It would be really great if each of these had a comment of what they're looking for, and what they would transform to, especially the final `else`

e.g. `// Assert.IsTrue( foo == null ) -> Assert.IsNull( foo )`

Reply:
Done